### PR TITLE
add hold detection functionality

### PR DESCRIPTION
**Pull request by webloopbox**
**detectHold** function that handles the detection and completion of a hold action for both mouse and touch pointer types. 
It includes the following features:
- **hold detection**: this could be a right mouse click or a touch event. It sets up an interval to check the duration of the hold and listens for a pointer up event.

- **hold completion**: when a pointer up event occurs, the function terminates the hold detection process. If the hold duration was 200 milliseconds or more, it resolves a promise to indicate that the hold action has been completed.

### CODE REVIEW

File: src/types/Behavior.ts [hunk 1]
```diff
@@ -10,6 +10,8 @@ export type HandlerFn<TEvent extends React.SyntheticEvent | Event> = (
 ) => ReactGridStore;
 
 export type PointerEventHandler = HandlerFn<React.PointerEvent<HTMLDivElement> | PointerEvent>;
+export type PointerHoldEventHandler = HandlerFn<React.PointerEvent<HTMLDivElement> | PointerEvent>;
+export type PointerHoldTouchEventHandler = HandlerFn<React.PointerEvent<HTMLDivElement> | PointerEvent>;
```
Comment (Michota) [inline src/types/Behavior.ts]: Why would we add two another types that are literally the same as `PointerEventHandler`?

File: src/controllers/handlePointerDown.ts [hunk 2]
```diff
@@ -7,12 +7,23 @@ import { updateStoreWithApiAndEventHandler } from "../utils/updateStoreWithApiAn
 import { ReactGridStore } from "../types/ReactGridStore.ts";
 
 export const handlePointerDown = (event: React.PointerEvent<HTMLDivElement>, storeApi: StoreApi<ReactGridStore>) => {
+  const usedTouch = event.pointerType !== "mouse"; // * keep in mind there is also "pen" pointerType
+
+  let holdTimeoutId: NodeJS.Timeout;
+
+  if (event.button === 2 || usedTouch) {
```
Comment (Michota) [inline src/controllers/handlePointerDown.ts]: What is it?

**Review by Michota — CHANGES_REQUESTED** [review]
Small changes are needed.

File: src/controllers/handlePointerDown.ts (hunk 2 above)
Comment (webloopbox) [reply src/controllers/handlePointerDown.ts]: If the event is a right-click or a touch event, a timeout is set to trigger. ```holdTimeoutId``` is used to clear the timeout if a pointer up event occurs before 500 ms, so the hold handler function will not be called

Comment (Michota) [reply src/controllers/handlePointerDown.ts]: Okay. Can you create some sort of constant assertion or simply a normal const that will describe `event.button === 2`, so the code will be easier to understand?

Comment (webloopbox) [reply src/controllers/handlePointerDown.ts]: Yup, it's a good idea

**Review by Michota — APPROVED** [review]

**Pull request merged**